### DISCUSSION TITLE
Custom memory and cpu count for docker image

### DISCUSSION THREAD

**Discussion by rohit04saluja** [hellt/vrnetlab#263] [Q&A]
Docker images are built with fixed memory and CPU count that is defined in `launch.py`. However, I need to change the resources for different images based on deployment requirements and releases.

Is there a way to define memory and CPU at the time of docker image built without editing `launch.py` everytime?

**Comment by hellt** [comment]
you can use https://containerlab.dev/manual/vrnetlab/#tuning-qemu-parameters

**Reply by rohit04saluja** [reply]
Just to understand the methodology here, `vrnetlab` repo is to be used to create a base image. Further tuning should be done at runtime.

Is my understanding correct?

**Reply by hellt** [reply]
Correct
